### PR TITLE
LPS-196062 - [Add support for system ObjectActions] Integration Test

### PR DESCRIPTION
**Pull request by G-Sa**
[LPS-196062](https://liferay.atlassian.net/browse/LPS-196062)

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 1]
```diff
@@ -1338,6 +1339,31 @@ public void testAddObjectActionWithSystemObject() throws Exception {
 		_objectFieldLocalService.deleteObjectField(objectField3);
 	}
 
+	@Test
+	public void testDeleteObjectAction() throws Exception {
+		ObjectAction systemObjectAction =
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1352]: Could you delete this variable after the assertFailure, please?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 2]
```diff
@@ -1449,6 +1450,98 @@ public void testUpdateObjectAction() throws Exception {
...
+			LocalizedMapUtil.getLocalizedMap("Baker Error Message"),
+			LocalizedMapUtil.getLocalizedMap("Baker Label"), "Able",
+			systemObjectAction, ObjectActionExecutorConstants.KEY_GROOVY,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			UnicodePropertiesBuilder.put(
+				"secret", "30624700"
+			).put(
+				"url", "https://onafterdelete.com"
+			).build(),
+			ObjectActionConstants.STATUS_NEVER_RAN);
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1551]: Can we move this scenario after the Add/Assert ?
Then, this test would be:
- successful scenario
- failed scenario 

Like we did with object relationship test. What do you think?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 3]
```diff
@@ -1449,6 +1450,98 @@ public void testUpdateObjectAction() throws Exception {
...
+			systemObjectAction, ObjectActionExecutorConstants.KEY_WEBHOOK,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			UnicodePropertiesBuilder.put(
+				"secret", "0123456789"
+			).put(
+				"url", "https://onafteradd.com"
+			).build(),
+			ObjectActionConstants.STATUS_NEVER_RAN);
+
+		String liferayMode = SystemProperties.get("liferay.mode");
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1492]: Can you add a comment like we added in the Object Relationship test?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 4]
```diff
@@ -218,43 +225,43 @@ public void testAddObjectAction() throws Exception {
 			() -> _addObjectAction(
 				RandomTestUtil.randomString(), RandomTestUtil.randomString(),
 				RandomTestUtil.randomString(), name,
-				ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE));
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:219]: Are we going to test adding a system object action?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 5]
```diff
@@ -612,13 +625,24 @@ public void testAddObjectAction() throws Exception {
...
 
+		ObjectAction systemObjectAction = _addObjectAction(
+			name, ObjectActionExecutorConstants.KEY_WEBHOOK,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			UnicodePropertiesBuilder.put(
+				"secret", "onafteradd"
+			).put(
+				"url", "https://onafteradd.com"
+			).build(),
+			false);
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:636]: Shouldn't the last parameter be true?

Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:636]: I think we could add an Object Action and see that if the action is working.
Like the previous examples.

Comment (marcelabc) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:636]: we can talk if you want.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 6]
```diff
@@ -1451,6 +1452,87 @@ public void testUpdateObjectAction() throws Exception {
 				"url", "https://onafterdelete.com"
 			).build(),
 			ObjectActionConstants.STATUS_NEVER_RAN);
+
+		String externalReferenceCode2 = RandomTestUtil.randomString();
+
+		ObjectAction systemObjectAction =
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1519]: Could you delete this variable `systemObjectAction` in the end of the test, please?

Comment (marcelabc) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1519]: You could delete `objectAction` too  :grimacing:

**Review by marcelabc — COMMENTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by G-Sa** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by marcelabc** [comment]
ci:reevaluate:1287401_8024

**Comment by G-Sa** [comment]
ci:test:relevant

**Comment by G-Sa** [comment]
ci:test:relevant

**Comment by G-Sa** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Hey @G-Sa 
Could you analyze the CI? If you need help, ask the QA team please.

**Comment by marcelabc** [comment]
Manually forwarded: https://github.com/brianchandotcom/liferay-portal/pull/142854

**Pull request closed without merging**